### PR TITLE
[nginx-ingress-controller] Readiness probe that works behind a CP lb

### LINKED ISSUE #1507: Figure out readiness probe for nginx ingress that works behind a CP lb
Flesh out the details discussed in: https://github.com/kubernetes/contrib/pull/1501#discussion-diff-74324489 and update the probes accordingly.

### PR DESCRIPTION
**Pull request by aledbf**
fixes #1507

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1749)

### CODE REVIEW

File: ingress/controllers/nginx/nginx.tmpl [hunk 1]
```diff
@@ -265,6 +265,12 @@ http {
         {{ end }}
 
         {{ if eq $server.Name "_" }}
+        # health checks in cloud providers require the use of port 80
+        location /ingress-controller-healthz {
```
Comment (bprashanth) [inline ingress/controllers/nginx/nginx.tmpl]: can you make this the default, and allow people to config it through yaml?

**Review by bprashanth — COMMENTED** [review]
awesome! thanks just the one nit

File: ingress/controllers/nginx/nginx.tmpl (hunk 1 above)
Comment (aledbf) [inline ingress/controllers/nginx/nginx.tmpl]: Using a flag with `/ingress-controller-healthz` as default is ok?

Comment (bprashanth) [inline ingress/controllers/nginx/nginx.tmpl]: any reason it shouldn't be a flag? if i have an ingress controller behind an ingress controller how would i modify it :)

**Comment by bprashanth** [comment]
Thanks LGTM, modifying title to note have wip

**Comment by bprashanth** [comment]
oh please add a todo to scrub incoming ingresses for the url passed through the arg (can come laters, or just file a bug so we can track it)

**Comment by aledbf** [comment]
> file a bug so we can track it)

done #1754

**Comment by test-foxish** [comment]
recomputing cla status...

**Pull request merged**